airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/pipeline.py

```diff
@@ -431,9 +478,39 @@ async def _run(self) -> StepResult:
         return await super()._run()
 
 
+class CheckConnectorVersionIncrement(Step):
```

Comment:
Can you add a docstring explaining that this particular check gates the publishing execution based on whether the connector version has been incremented, to avoid failures and re-writing existing connector container image with a different container under the same version tag? Add a note about respecting the github env var that allows us to override this check.